src/js_native_api_v8.cc

```diff
@@ -2997,3 +2997,38 @@ napi_status napi_get_instance_data(napi_env env,
 
   return napi_clear_last_error(env);
 }
+
+napi_status napi_arraybuffer_detach(napi_env env,
+                                     napi_value arraybuffer) {
+  CHECK_ENV(env);
+  CHECK_ARG(env, arraybuffer);
+
+  v8::Local<v8::Value> value = v8impl::V8LocalValueFromJsValue(arraybuffer);
+  if (!value->IsArrayBuffer()) {
+    napi_throw_type_error(
+        env,
+        "ERR_NAPI_INVALID_ARRAYBUFFER_ARGS",
+        "Expects an arraybuffer");
+    return napi_set_last_error(env, napi_invalid_arg);
+  }
+
+  v8::Local<v8::ArrayBuffer> it = value.As<v8::ArrayBuffer>();
+  if (!it->IsExternal()) {
+    napi_throw_type_error(
+        env,
+        "ERR_NAPI_INVALID_ARRAYBUFFER_ARGS",
```

Comment:
If we do decide to throw here and below then the code for the two errors should be different, because two different messages should not be associated with the same code.

Reply:
Maybe change the code to `ERR_NAPI_ARRAYBUFFER_NOT_DETACHABLE`.